common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -7,37 +7,42 @@ import constants = require('./constants');
 import path = require('path');
 import fs = require('fs');
 import jwt = require('jsonwebtoken');
+import msal = require('@azure/msal-node');
+import crypto = require("crypto");
 
 tl.setResourcePath(path.join(__dirname, 'module.json'), true);
 
 export class ApplicationTokenCredentials {
-    private clientId: string;
-    private domain: string;
-    private authType: string;
-    private secret?: string;
-    private accessToken?: string;
-    private certFilePath?: string;
-    private isADFSEnabled?: boolean;
     public baseUrl: string;
     public authorityUrl: string;
     public activeDirectoryResourceId: string;
     public isAzureStackEnvironment: boolean;
     public scheme: number;
     public msiClientId: string;
+
+    private clientId: string;
+    private tenantID: string;
+    private authType: string;
+    private secret?: string;
+    private accessToken?: string;
+    private certFilePath?: string;
+    private isADFSEnabled?: boolean;
     private token_deferred: Q.Promise<string>;
+    private useMSAL: boolean;
+    private msalInstance: msal.ConfidentialClientApplication;
 
-    constructor(clientId: string, domain: string, secret: string, baseUrl: string, authorityUrl: string, activeDirectoryResourceId: string, isAzureStackEnvironment: boolean, scheme?: string, msiClientId?: string, authType?: string, certFilePath?: string, isADFSEnabled?: boolean, access_token?: string) {
+    constructor(clientId: string, tenantID: string, secret: string, baseUrl: string, authorityUrl: string, activeDirectoryResourceId: string, isAzureStackEnvironment: boolean, scheme?: string, msiClientId?: string, authType?: string, certFilePath?: string, isADFSEnabled?: boolean, access_token?: string, useMSAL?: boolean) {
```

Comment:
Why no tenantId?

Reply:
AzureModel file contains lots of "ID" definition, that's why sometimes I've used ID syntax but for the file consistency I've changed them to "Id"